src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.Current.net46.cs

```diff
@@ -27,7 +27,9 @@ public partial class Activity
 
         private static void SetCurrent(Activity? activity)
         {
+            Activity? previous = s_current.Value;
             s_current.Value = activity;
+            CurrentChanged?.Invoke(null, new ActivityChangedEventArgs(previous, activity));
```

Comment:
This means that arbitrary user code is now going to run anywhere Activity.Current is set.  Is that always going to be ok? e.g. might that ever be set while holding a lock?

Reply:
It is possible there will be a lock held so it isn't foolproof, but its roughly the same type of problem that people already deal with successfully in ActivityListener. The small set of telemetry libraries that hook up to these mechanisms need to be careful to keep their dependencies limited both to avoid creating functional or performance problems.